Firebase/ios/BonfireTests/Chat/PersistedChatServiceTests.swift

```diff
@@ -0,0 +1,105 @@
+import XCTest
+import RxSwift
+import RxTests
+@testable import Bonfire
+
+class PersistedChatServiceTests: XCTestCase {
+
+    let channel =  Channel(name: "TestChannel", access: .Public)
+
+    var scheduler: TestScheduler!
+    var testableObserver: TestableObserver<DatabaseResult<Chat>>!
+
+    override func setUp() {
+        super.setUp()
+        scheduler = TestScheduler(initialClock: 0)
+        testableObserver = scheduler.createObserver(DatabaseResult<Chat>)
+    }
+
+
+    struct TestErrorType: ErrorType {}
+
+    struct MockChatDatabase: ChatDatabase {
+        let chatObservable: Observable<Chat>
+        let sendMessageExpectation: ((Message, Channel) -> Void)?
+
+        init(chatObservable: Observable<Chat>,
+             sendMessageExpectation: ((Message, Channel) -> Void)? = nil) {
+            self.chatObservable = chatObservable
+            self.sendMessageExpectation = sendMessageExpectation
+        }
+
+        func chat(channel: Channel) -> Observable<Chat> {
+            return chatObservable
+        }
+
+        func sendMessage(message: Message, channel: Channel) {
+            sendMessageExpectation?(message, channel)
+        }
+    }
+
+    func testThatItMapsChatObservableToDatabaseResultObservable() {
+        // Given
+        let chat = Chat(channel: channel, messages: [])
+
+        let testChatEvents = scheduler.singleEventAndHang(chat)
+
+        let chatDatabase = MockChatDatabase(chatObservable: testChatEvents)
+        let chatService = PersistedChatService(chatDatabase: chatDatabase)
+
+        // When
+        chatService.chat(channel).subscribe(testableObserver)
+        scheduler.start()
+
+        // Then
+        let expectedEvents = [
+            next(0, DatabaseResult.Success(chat))
+        ]
+
+        XCTAssertEqual(testableObserver.events, expectedEvents)
```

Comment:
I like this quite a bit - being able to define clearly the events you expect is neat. RxTests FTW.